diff --git a/video/out/gpu/shader_cache.c b/video/out/gpu/shader_cache.c
index bf0983f5f0..57e8edb18f 100644
--- a/video/out/gpu/shader_cache.c
+++ b/video/out/gpu/shader_cache.c
@@ -17,7 +17,7 @@
 #include "utils.h"
 
 // Force cache flush if more than this number of shaders is created.
-#define SC_MAX_ENTRIES 256
+#define SC_MAX_ENTRIES 1024
 
 union uniform_val {
     float f[9];         // RA_VARTYPE_FLOAT
