diff --git a/src/intel/vulkan/meson.build b/src/intel/vulkan/meson.build
index 1886ea0904a..93332c4bf99 100644
--- a/src/intel/vulkan/meson.build
+++ b/src/intel/vulkan/meson.build
@@ -82,6 +82,19 @@
 # Deprecated: replaced by VK_DRIVER_FILES above
 devenv.append('VK_ICD_FILENAMES', _dev_icd.full_path())
 
+libanv_per_hw_ver_libs_deps = [dep_libdrm,  
+  dep_valgrind, idep_nir_headers, idep_genxml,
+  idep_vulkan_util_headers, idep_vulkan_wsi_headers,
+  idep_vulkan_runtime_headers, idep_mesautil,
+  idep_intel_driver_ds_headers,
+  idep_intel_shaders, idep_intel_blorp,]
+  
+
+
+if with_platform_wayland
+  libanv_per_hw_ver_libs_deps += dep_wayland_client
+endif
+
 libanv_per_hw_ver_libs = []
 anv_per_hw_ver_files = files(
   'genX_blorp_exec.c',
@@ -115,13 +115,7 @@
     ],
     c_args : anv_flags + ['-DGFX_VERx10=@0@'.format(_gfx_ver)],
     gnu_symbol_visibility : 'hidden',
-    dependencies : [
-      dep_libdrm, dep_valgrind, idep_nir_headers, idep_genxml,
-      idep_vulkan_util_headers, idep_vulkan_wsi_headers,
-      idep_vulkan_runtime_headers, idep_mesautil,
-      idep_intel_dev, idep_intel_driver_ds_headers,
-      idep_intel_shaders, idep_intel_blorp,
-    ],
+    dependencies : libanv_per_hw_ver_libs_deps,
   )
 endforeach

