diff --git a/src/gallium/frontends/lavapipe/lvp_private.h b/src/gallium/frontends/lavapipe/lvp_private.h
index 81b85cc152a..ee053bf66b4 100644
--- a/src/gallium/frontends/lavapipe/lvp_private.h
+++ b/src/gallium/frontends/lavapipe/lvp_private.h
@@ -490,10 +490,6 @@ static inline uint32_t
 lvp_pipeline_types_from_shader_stages(VkShaderStageFlags stageFlags)
 {
    uint32_t types = 0;
-#ifdef VK_ENABLE_BETA_EXTENSIONS
-   if (stageFlags & MESA_VK_SHADER_STAGE_WORKGRAPH_HACK_BIT_FIXME)
-      types |= BITFIELD_BIT(LVP_PIPELINE_EXEC_GRAPH);
-#endif
    if (stageFlags & LVP_RAY_TRACING_STAGES)
       types |= BITFIELD_BIT(LVP_PIPELINE_RAY_TRACING);
    if (stageFlags & VK_SHADER_STAGE_COMPUTE_BIT)
