diff --git a/dlls/kernelbase/thread.c b/dlls/kernelbase/thread.c
index 12e1777262a..cc2a629483a 100644
--- a/dlls/kernelbase/thread.c
+++ b/dlls/kernelbase/thread.c
@@ -340,22 +340,18 @@
  */
 HANDLE WINAPI DECLSPEC_HOTPATCH OpenThread( DWORD access, BOOL inherit, DWORD id )
 {
-    HANDLE handle;
+    HANDLE handle = NULL;
     OBJECT_ATTRIBUTES attr;
     CLIENT_ID cid;
 
-    attr.Length = sizeof(attr);
-    attr.RootDirectory = 0;
+    memset(&attr, 0, sizeof(attr));
+    attr.Length = sizeof(OBJECT_ATTRIBUTES);
+    attr.RootDirectory = NULL;
     attr.Attributes = inherit ? OBJ_INHERIT : 0;
-    attr.ObjectName = NULL;
-    attr.SecurityDescriptor = NULL;
-    attr.SecurityQualityOfService = NULL;
 
-    cid.UniqueProcess = 0;
-    cid.UniqueThread = ULongToHandle( id );
-
-    if (!set_ntstatus( NtOpenThread( &handle, access, &attr, &cid ))) handle = 0;
-    return handle;
+    memset(&cid, 0, sizeof(cid));
+    cid.UniqueThread = (HANDLE)(ULONG_PTR)id;
+    return set_ntstatus(NtOpenThread(&handle, access, &attr, &cid)) ? handle : NULL;
 }
